builtInFunctions/esdtDataStorage.go

```diff
@@ -143,7 +143,7 @@ func (e *esdtDataStorage) getESDTNFTTokenOnDestinationWithAccountsAdapterOptions
 		return nil, false, err
 	}
 
-	if !e.enableEpochsHandler.IsFlagEnabled(SaveToSystemAccountFlag) || nonce == 0 {
+	if !e.enableEpochsHandler.IsFlagEnabled(SaveToSystemAccountFlag) || nonce == 0 || isNFTWithMetadataOnAccount(esdtData) {
```

Comment:
Yes, because inside `isNFTWithMetadataOnAccount()` we check 
	```if esdtToken.Type != uint32(core.NonFungibleV2) {
		return false
	}```

core.NonFungibleV2 will be added on accounts only when this feature is merged